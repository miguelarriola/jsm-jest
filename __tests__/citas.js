import Citas from '../js/classes/Citas';

describe('Probar laclase de citas', () => {
  const citas = new Citas();
  const id = Date.now();

  test('Agregar una nueva cita', () => {
    const citaObj = {
      id,
      mascota: 'Hook',
      propietario: 'Juan',
      telefono: '546353323',
      fecha: '10-12-2020',
      hora: '10:30',
      sintomas: 'Solo duerme',
    };
    citas.agregarCita(citaObj);
    expect(citas).toMatchSnapshot();
  });

  test('Actualizar cita', () => {
    const citaActualizada = {
      id,
      mascota: 'Nuevo Nombre',
      propietario: 'Juan',
      telefono: '546353323',
      fecha: '10-12-2020',
      hora: '10:30',
      sintomas: 'Solo duerme',
    };
    citas.agregarCita(citaActualizada);
    expect(citas).toMatchSnapshot();
  });

  test('Eliminar cita', () => {
    citas.eliminarCita(id);
    expect(citas).toMatchSnapshot();
  });
});
