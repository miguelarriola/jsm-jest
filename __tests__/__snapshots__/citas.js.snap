// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar laclase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1606709911545,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "546353323",
    },
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1606709911545,
      "mascota": "Nuevo Nombre",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "546353323",
    },
  ],
}
`;

exports[`Probar laclase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1606709911545,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "546353323",
    },
  ],
}
`;

exports[`Probar laclase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
